src/utils.jl

```diff
@@ -40,6 +41,23 @@ function bhattacharyya(x::kmerdist,y::kmerdist)
     return 1/4*log(arg1 + arg2)
 end
 
+"""
+a function that computes the Bhattacharyya coefficient of
+n-distribution
+"""
+function multi_bhattacharyya(data::Array{Array{Float32}})
+    value = 0
```

Comment:
1. Default parameter of n::Union{Nothing, Int} = nothing
2. data should be here Array(Kmerdist} objects for single column in our multiple sequnce alignment.
3. If n = nothing, then n = as described previously
4. n is number of partitions. Extract min and maximum from ALL data points.
5. Create n partitions between min and max.
6. For each of the partitions starting from min:first_partition_end COUNT number of datapoints that fall into this partition for each of the kmerdist - call this our p, q, l
7. then do (p*q*l) ^ 1/length(data)

Reply:
Could you further elaborate on point 6? How do you end up with 3 variables from counting a number of datapoints?

Reply:
We want to see how much distributions overlap, a sort of union of their overlap is our ideal output.
Instead of computing integral, we just create bins/partitions and see what data points fall into each partition.
Example:
a = [1, 2, 3, 4, 5]
b = [2, 3, 4, 6, 7]
c = [2, 3, 6, 8, 9]
we have k = 3 datasets, we make n = ceil( (5 + 5 + 5) / 3) = 5
min = 1
max = 9
n partitions = [1.0, 2.6), [2.6, 4.2), [4.2, 5.8), [5.8, 7.4), [7.4, 9.0]
From a into first partition falls 2 data points, b has 1, c has 1
(2 * 1 * 1) ^ 1/3
Repeat for other partitions and sum

Reply:
Make this function as fast as possible as this will be the heart of our calculations